charts/kube-prometheus-stack/hack/sync_prometheus_rules.py

```diff
@@ -160,7 +160,10 @@ def new_representer(dumper, data):
         'replacement': '$1',
         'init': ''},
     'job="kube-state-metrics"': {
-        'replacement': 'job="kube-state-metrics", namespace=~"{{ $targetNamespace }}"',
+        'replacement': 'job="{{ $kubeStateMetricsJob }}"',
+        'init': '{{- $kubeStateMetricsJob := include "kube-prometheus-stack-kube-state-metrics.fullname" . }}'},
+    'job="{{ $kubeStateMetricsJob }}"': {
+        'replacement': 'job="{{ $kubeStateMetricsJob }}", namespace=~"{{ $targetNamespace }}"',
```

Comment:
this seems to be unused, or do I miss something?

Reply:
This is still used in [kubernetes-apps.yaml](https://github.com/prometheus-community/helm-charts/pull/4160/files#diff-742cb8cc85db84d4b08b8680cfd7a4518869cb5e777c991639806a80dff2d25aR41) if I am not mistaken.